app/src/main/java/org/schabi/newpipe/util/external_communication/TimestampExtractor.java

```diff
@@ -0,0 +1,79 @@
+package org.schabi.newpipe.util.external_communication;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Extracts timestamps.
+ */
+public final class TimestampExtractor {
+    public static final Pattern TIMESTAMPS_PATTERN = Pattern.compile(
+            "(?:^|(?!:)\\W)(?:([0-5]?[0-9]):)?([0-5]?[0-9]):([0-5][0-9])(?=$|(?!:)\\W)");
```

Comment:
Why stop with hours at `60`? I'd just make that illimitate: on a video 100 hours long, how could one target the 90th hour (e.g. 90:32:11)
```suggestion
            "(?:^|(?!:)\\W)(?:([0-9]+):)?([0-5]?[0-9]):([0-5][0-9])(?=$|(?!:)\\W)");
```

Reply:
@Stypox ~illimitate~ limitless

Reply:
> Why stop with hours at `60`? I'd just make that ~illimitate~ limitless: on a video 100 hours long, how could one target the 90th hour (e.g. 90:32:11)

Some things to consider:
* Don't ask me I just copied it 😆, ask the inventor → https://github.com/TeamNewPipe/NewPipe/pull/6851#issuecomment-894060228
* Such a solution will throw an error if it exceeded the int limit
* Video lengths are capped on most platforms:
  * YouTube: max 12h
  Source: https://support.google.com/youtube/answer/71673
  * SonarCloud: max 6h 
  Source: https://help.soundcloud.com/hc/en-us/articles/115003452847-Uploading-requirements
  * Media CCC: ?
  * PeerTube: depends on the instance; clicked through a few instances and most of the have a contingent of ~20GB which are ~31h in average resolution video (I'm posting no links regarding peer tube instances, most of whom I visited host hoax videos - even though I searched with "Sensitive videos: blur")
  * Bandcamp: 600MB; mp3 min bitrate =96Kbps (according to [adobe](https://www.adobe.com/creativecloud/video/discover/audio-bitrate.html)) → 600MB / 96Kbps / 8 = ~781s = ~13h
  Source: https://get.bandcamp.help/hc/en-us/articles/360007901993-What-s-the-maximum-upload-size-

According to https://www.statista.com/statistics/1026923/youtube-video-category-average-length/ the average YT video has a length of ~12minutes. I myself rarely see/noticed videos which are longer than 1-2h.

According to the above the case of navigating to a video timestamp which is >60h is nearly zero. 
If there is such a case, it currently [simply doesn't detect the timestamp](https://regex101.com/r/EWQG8D/1) and the user has to move there by himself.

So I wouldn't change anything for now. 
If someone will ever have that problem, the person can open an issue.

Reply:
Ok, you are right